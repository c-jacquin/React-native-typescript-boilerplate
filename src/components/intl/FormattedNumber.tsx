import React, { StatelessComponent } from 'react'
import { FormattedNumber } from 'react-intl'
import { Text } from 'react-native'

import { IFormattedNumberProps } from './types'

const NativeFormattedNumber: StatelessComponent<IFormattedNumberProps> = ({
    style,
    formatStyle,
    ...other,
}) => {
    const formatOptions = {
        ...other,
        style: formatStyle,
    }

    return (
        <FormattedNumber {...formatOptions}>
            {(localized: string) => <Text style={style}>{localized}</Text>}
        </FormattedNumber>
    )
}

export default NativeFormattedNumber
