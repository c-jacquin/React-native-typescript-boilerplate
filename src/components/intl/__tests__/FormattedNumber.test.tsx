import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { IntlProvider } from 'react-intl'

import FormattedNumber from '../FormattedNumber'

describe('FormattedNumber', () => {
    it('should renders correctly', () => {
        const tree = renderer
            .create(
                <IntlProvider locale="en">
                    <FormattedNumber style={{}} value={'foo'} />
                </IntlProvider>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
