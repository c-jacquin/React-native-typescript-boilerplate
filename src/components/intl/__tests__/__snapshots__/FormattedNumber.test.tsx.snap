// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormattedNumber should renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}
>
  NaN
</Text>
`;
